common-testing/src/emulator.rs

```diff
@@ -99,7 +99,29 @@ pub fn parse_output<T: DeserializeOwned>(
 }
 
 /// Create a temporary directory with a new Cargo project that has nexus_rt as a local dependency.
-pub fn setup_guest_project(runtime_path: &PathBuf) -> TempDir {
+fn get_runtime_path() -> PathBuf {
+    // Try to find the runtime directory by looking for Cargo.toml
+    let current_dir = std::env::current_dir().unwrap();
+    let mut path = current_dir.clone();
+
+    // Look for runtime directory in current and parent directories
+    for _ in 0..5 {
```

Comment:
Is there a way to make this more "principled" then going an arbitrary depth deep to try and find a valid path? I'm open to trying to provide a measure of "best effort" around this problem, but I'd like to see something a bit more well-formed in terms of an approach.